hybris/egl/egl.c

```diff
@@ -139,9 +171,69 @@ struct _EGLDisplay *hybris_egl_display_get_mapping(EGLDisplay display)
 	return EGL_NO_DISPLAY;
 }
 
-EGLDisplay eglGetDisplay(EGLNativeDisplayType display_id)
+static const char * _defaultEglPlatform()
+{
+	char *egl_platform;
+
+	// Mesa uses EGL_PLATFORM for its own purposes.
+	// Add HYBRIS_EGLPLATFORM to avoid the conflicts
+	egl_platform = getenv("HYBRIS_EGLPLATFORM");
+
+	if (egl_platform == NULL)
+		egl_platform = getenv("EGL_PLATFORM");
+
+	// The env variables may be defined yet empty
+	if (egl_platform == NULL || strcmp(egl_platform, "") == 0)
+		egl_platform = DEFAULT_EGL_PLATFORM;
+
+	return egl_platform;
+}
+
+#ifndef WANT_GLVND
+static
+#endif
+EGLDisplay __eglHybrisGetPlatformDisplayCommon(EGLenum platform,
+        void *display_id, const EGLAttrib *attrib_list)
 {
+	// We have nothing to do with attrib_list at the moment. Silence the unused
+	// variable warning.
+	(void) attrib_list;
+
 	HYBRIS_DLSYSM(egl, &_eglGetDisplay, "eglGetDisplay");
+
+	if (!_eglGetDisplay) {
+		__eglHybrisSetError(EGL_NOT_INITIALIZED);
+		return EGL_NO_DISPLAY;
+	}
+
+	const char * hybris_ws;
+	switch (platform) {
+		case EGL_NONE:
+			hybris_ws = _defaultEglPlatform();
+			break;
+
+		case EGL_PLATFORM_ANDROID_KHR:
+			// "null" ws passthrough everything, which essentially means
+			// the Android platform. Beware confustion with NULL value.
```

Comment:
Typo: confustion should be confusion.

Reply:
I've reworded it completely. There shouldn't be a typo this time, because the phrase is copied from Wikipedia's template. :D